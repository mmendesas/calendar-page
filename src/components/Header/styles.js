import styled from 'styled-components';

export const Container = styled.header`
  height: 50px;
  padding: 0 30px;
  background: #fff;
  color: #58636d;
  border-bottom: 1px solid #b8c3cc;

  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Menu = styled.ul`
  list-style: none;

  a {
    font-size: 16px;
    text-decoration: none;
    padding: 15px;
    color: #58636d;
    transition: box-shadow 0.5s ease 0s, text-shadow 0.5s ease 0s;

    &:hover {
      box-shadow: #0675ce 0px -5px 0px 0px inset;
      text-shadow: #58636d 0px 0px 0.1px, #58636d 0px 0px 0.1px;
    }
  }
`;

export const Avatar = styled.div`
  background: url(https://vanhackblobstorageprod.blob.core.windows.net/img/User/139842.jpg)
    no-repeat center;
  background-size: cover;
  width: 35px;
  height: 35px;
  border-radius: 50%;
  border: 3px solid #eee;
  transition: border 0.5s ease 0s;

  &:hover {
    border: 3px solid #ccc;
  }
`;

export const UserInfo = styled.a`
  padding-left: 50px;
  display: flex;
  justify-content: center;
`;
