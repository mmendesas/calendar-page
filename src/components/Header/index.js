import React from 'react';
import { Link } from 'react-router-dom';

import { Container, Menu, UserInfo, Avatar } from './styles';

import logo from '../../assets/images/vanhack-logo.svg';

export default function index() {
  return (
    <Container>
      <Link to="/">
        <img src={logo} alt="Logo" />
      </Link>
      <Menu>
        <Link to="/jobs">Jobs</Link>
        <Link to="/events">Events</Link>
      </Menu>
      <UserInfo href="https://github.com/mmendesas">
        <Avatar />
      </UserInfo>
    </Container>
  );
}
