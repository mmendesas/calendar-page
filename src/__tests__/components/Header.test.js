import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter as Router } from 'react-router-dom';
import Header from '~/components/Header';

describe('[Component] Header', () => {
  const setup = () => {
    const utils = render(
      <Router>
        <Header />
      </Router>
    );

    return {
      utils,
    };
  };

  it('should be match snapshot', () => {
    const { utils } = setup();
    const { getByTestId } = utils;
    expect(getByTestId('header')).toMatchSnapshot();
  });

  it('should render Header with link to events page', () => {
    const { utils } = setup();
    const { getByText, debug } = utils;

    expect(getByText('Events')).toHaveProperty(
      'href',
      'http://localhost/events'
    );
  });
});
