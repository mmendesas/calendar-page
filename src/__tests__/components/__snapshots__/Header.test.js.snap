// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Header should be match snapshot 1`] = `
<header
  class="sc-bdVaJa eEVJUI"
  data-testid="header"
>
  <a
    href="/"
  >
    <img
      alt="Logo"
      src="vanhack-logo.svg"
    />
  </a>
  <ul
    class="sc-bwzfXH cFpKRr"
  >
    <a
      href="/jobs"
    >
      Jobs
    </a>
    <a
      href="/events"
    >
      Events
    </a>
  </ul>
  <a
    class="sc-bxivhb jUXNXE"
    href="https://github.com/mmendesas"
  >
    <div
      class="sc-htpNat hmzhmz"
    />
  </a>
</header>
`;
